src/dynamicprompts/samplers/cycle.py

```diff
@@ -38,8 +37,8 @@ def _get_combination_samples(
         else:
             c_1, c_rest = combo[0], combo[1:]
 
-            gen = sampler_router.generator_from_command(c_1)
-            rest_gen = _get_combination_samples(c_rest, sampler_router)
+            gen = sampling_context.generator_from_command(c_1)
+            rest_gen = _get_combination_samples(c_rest, sampling_context)
             arrs = get_arrs(gen, rest_gen)
```

Comment:
As you mentioned when you first reviewed this PR, `get_arrs` is an awful name for a function. I think the concept that needs to be extracted here is what I suggested in combinatorial.py with these functions:

```python
def sample_with_nonfinite_command(nonfinite_command, other_commands) -> Iterable[list_of_tokens]
```
and
```python
def sample_with_finite_command(finite_command, other_commands) -> Iterable[list_of_tokens]
```
essentially these functions abstract how we sample lists of tokens from collections of finite and non-finite commands. Extract this logic into functions will make it easier to read both combinatorial.py and cycle.py